src/widget/weffectselector.cpp

```diff
@@ -0,0 +1,94 @@
+#include <QtDebug>
+
+#include "widget/weffectselector.h"
+
+#include "effects/effectsmanager.h"
+#include "widget/effectwidgetutils.h"
+
+namespace {
+    const QString nullEffectId = "NULL";
+}
+
+WEffectSelector::WEffectSelector(QWidget* pParent, EffectsManager* pEffectsManager)
+        : QComboBox(pParent),
+          WBaseWidget(pParent),
+          m_pEffectsManager(pEffectsManager) {
+
+    setInsertPolicy(QComboBox::InsertAlphabetically);
```

Comment:
It turns out this only applies to user-entered data, not programmatically entered data -- so you can remove it.